libs/s25main/ai/aijh/AIConstruction.cpp

```diff
@@ -550,9 +550,7 @@ BuildingType AIConstruction::ChooseMilitaryBuilding(const MapPoint pt)
                 bld = BLD_CATAPULT;
             else
             {
-                if(randmil % 2 == 0)
-                    bld = biggestBld;
-                else if(aii.CanBuildBuildingtype(BLD_WATCHTOWER))
+                if(randmil % 2 == 0 && aii.CanBuildBuildingtype(BLD_WATCHTOWER))
```

Comment:
thats wrong?

Reply:
Yes and no. `randmil % 2 == 0` is semantically: with a 50% chance. I just changed the "sides of the coin"
Check the whole if-elseif-else chain to verify that it is correct

Reply:
hmm okay yes you're right. but changing that will break existing replays

Reply:
This is AI. AI decisions are not part of the replay, only the GCs